proposals/cleanup.md

```diff
@@ -233,6 +233,87 @@ Kyverno could alternatively leverage a CronJob resource to perform the deletions
 | CronJob will do both: schedule the deletion and execute the deletion. | A CronJob created through the cleanup rule should be ignored by other rules(validate and mutate). For example, A validate policy should not block the creation of that CronJob. This can cause some complexities in the approach. We can label these CronJobs and ignore them while applying other policies. |
 | The CronJobs can be easily removed from the cluster after the completion by setting their ownerReference to the resource it was created to delete. |  |
 
+## Finalized Implementation
+
+Finalized implementation is **Implementation 2 (reuse the Kubernetes CronJob resource)**. So, we will create new CRDs **CleanupPolicy** and **ClusterCleanupPolicy** for handling resource cleanup. 
+
+CleanupPolicy CRD manfiests example:
+```
+apiVersion: kyverno.io/v1alpha1
+kind: CleanupPolicy
+metadata:
+  name: cleanuppolicy
+  namespace: default
+spec:
+  match:
+    any:
+    - resources:
+        kinds:
+          - Pod
+  conditions:
+    any:
+    - key: "{{ request.name }}"
+      operator: Equals
+      value: example
+  schedule: "* * * * *"
+```
+A cleanup controller will reconcile these CRDs and create CronJobs which will execute commands to delete the specified resources. For implementing cleanup controller we have 2 options:
+
+
+### 1. Creating CronJobs per resource.
+A cleanup controller will be creating a CronJob for each of the matching resources. CronJob will run the kubectl command to delete these resources. CronJob will run according to the schedule specified in the cleanupPolicy. Creation and updation of a new cleanup policy will trigger the reconciliation and the reconciliation logic will get all the matching resources and will create a CronJob per resource which will have info and commands to delete the matching resource.
```

Comment:
"A cleanup controller will be creating" ->..will **create** a CronJob..

"CronJob will run according to the schedule specified in the cleanupPolicy." -> The CronJob is executed at the given schedule specified in the cleanup policy.

If this is a brief summary of the solution, you can remove "and the reconciliation logic will get all the matching resources and will create a CronJob per resource which will have info and commands to delete the matching resource.", this is detailed implementation.